### ISSUE TITLE
Query Error from iceberg external table when iceberg table with parquet format and lz4 compression 

### ISSUE THREAD

**Issue by tiannan-sr** [StarRocks/starrocks#3008]
### Steps to reproduce the behavior (Required)
1. create iceberg table with parquet format and lz4 compression-codec , insert data:
```
create table hive_prod.iceberg_db.iceberg_parquet_lz4(
 col_boolean boolean,
 col_int int,
 col_long long,
 col_float float,
 col_double double,
 col_decimal decimal(38,18),
 col_date date,
 col_timestamp timestamp,
 col_string string,
 col_binary binary,
 col_struct struct<a:string,b:int>,
 col_map map<string,int>
)
USING iceberg 
TBLPROPERTIES('write.format.default'='parquet','write.parquet.compression-codec'='lz4');

insert into hive_prod.iceberg_db.iceberg_parquet_lz4 select true,1,1.01,1.001,1.0001,10000000000.0000000001,cast('2020-01-01' as date),cast('2020-01-01 00:00:01' as timestamp),'Top 10 Unsolved Mysteries of Paleontological Dinosaurs, Did You Know?',cast('1110001010101011001001' as binary),null,null;
insert into hive_prod.iceberg_db.iceberg_parquet_lz4 select false,2,2.02,2.002,2.0002,20000000000.0000000002,cast('2020-02-01' as date),cast('2020-02-01 00:00:02' as timestamp),'Xi Jinping, general secretary of the Communist Party of China Central Committee, called on Tuesday for unrelenting efforts in exercising full and rigorous governance over the Party, saying that the CPC will continue to show zero tolerance for corruption.',cast('1110001010101011001001' as binary),null,null;
insert into hive_prod.iceberg_db.iceberg_parquet_lz4 select true,3,3.03,3.003,3.0003,30000000000.0000000003,cast('2020-03-01' as date),cast('2020-03-01 00:00:04' as timestamp),'Nation raises caution on overseas packages',cast('1110001010101011001001' as binary),null,null;
```

2. create iceberg external table and select:
```
mysql>  create external table ex_iceberg_tbl3 (
    ->  col_int int not null,
    ->  col_boolean boolean not null,
    ->  col_decimal decimal(38,18) not null,
    ->  col_long bigint not null,
    ->  col_float float not null comment "column float",
    ->  col_double double not null,
    ->  col_date date not null,
    ->  col_string varchar(200) not null comment "column string-varchar")
    ->  ENGINE=iceberg
    ->  properties (
    ->  "resource" = "iceberg_emr_tn",
    ->  "table" = "iceberg_parquet_lz4",
    ->  "database" = "iceberg_db");
Query OK, 0 rows affected (0.02 sec)

mysql> select * from ex_iceberg_tbl3;
ERROR 1064 (HY000): fail to do LZ4 decompress, error=-2
```

### Expected behavior (Required)
return right result

### Real behavior (Required)
return error: ERROR 1064 (HY000): fail to do LZ4 decompress, error=-2

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```
mysql> select current_version();
+------------------------+
| current_version()      |
+------------------------+
| QA_TEST_MASTER d404b84 |
+------------------------+
1 row in set (0.01 sec)
```

**Comment by caneGuy** [comment]
please assign to me thanks

**Comment by caneGuy** [comment]
this is related with lz4 compression, may be a parquet issue?i will debug this today

**Comment by tiannan-sr** [comment]
The problem has been fixed and merged into version branch-2.1 and main
